WalletWasabi/JsonConverters/NetworkJsonConverterNg.cs

```diff
@@ -0,0 +1,57 @@
+using NBitcoin;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace WalletWasabi.JsonConverters;
+
+/// <summary>
+/// Converter used to convert <see cref="Network"/> to and from JSON.
+/// </summary>
+/// <seealso cref="JsonConverter" />
+public class NetworkJsonConverterNg : JsonConverter<Network>
+{
+	/// <inheritdoc/>
+	public override bool HandleNull => true;
+
+	/// <inheritdoc/>
+	public override Network? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
We can get rid of the nullability of the return type if at the end we do a check like this.
```
Network network = Network.GetNetwork(networkString);
if (network is null) => throw Exception
return network
```

Reply:
I figured out that changing the nullability breaks the test. 
If we don't allow to return `null`, I think we change the behavior, too. Which goes against the goal of this PR.

It's up to you guys, if you want to deal with this in this PR or in another PR or never.